src/components/toggle/toggle.directive.ts

```diff
@@ -20,57 +22,86 @@ import type { IgcToggleController } from './toggle.controller.js';
 /* blazorSuppressComponent */
 export class IgcToggleDirective extends Directive {
   private part: PartInfo;
-  private shiftOptions? = {
-    mainAxis: true,
-  };
 
-  private floatingElement!: HTMLElement;
+  private _floatingElement!: HTMLElement;
+  private _dispose!: Function;
 
-  private updatePosition(
-    target: HTMLElement,
-    options: IgcToggleOptions,
-    controller?: IgcToggleController
-  ) {
-    this.floatingElement = this.createFloatingElement(options.open);
+  private target!: HTMLElement;
+  private options!: IgcToggleOptions;
+  private controller?: IgcToggleController;
 
-    if (!target) {
-      if (controller) {
-        controller.rendered = Promise.resolve();
-      }
-      return noChange;
+  private get floatingElement() {
+    if (!this._floatingElement) {
+      this._floatingElement = (this.part as ElementPart).element as HTMLElement;
     }
 
-    const promise = computePosition(target, this.floatingElement, {
-      placement: options.placement ?? 'bottom-start',
-      strategy: options.positionStrategy ?? 'absolute',
-      middleware: this.createMiddleware(options),
-    }).then(({ x, y }) => {
-      Object.assign(this.floatingElement.style, {
-        left: 0,
-        top: 0,
-        transform: `translate(${x}px,${y}px)`,
-      });
+    Object.assign(this._floatingElement.style, {
```

Comment:
Kind of an odd place to have this, assuming from the old code only `updatePosition` needs to switch open state _once_. Probably no noticeable hit to speak of, but still odd. Also bit funky the element needs to be cached, considering the part appears to be a static constructor param so likely just property access `this.part.element` (with type asserts) can do just fine